htlcswitch/hop/payload_test.go

```diff
@@ -356,6 +356,51 @@ var decodePayloadTests = []decodePayloadTest{
 		},
 		shouldHaveTotalAmt: true,
 	},
+	{
+		name:       "final blinded hop with total amount",
+		isFinalHop: true,
+		payload: []byte{
+			// amount
+			0x02, 0x00,
+			// cltv
+			0x04, 0x00,
+			// encrypted data
+			0x0a, 0x03, 0x03, 0x02, 0x01,
+		},
+		shouldHaveEncData: true,
+	},
```

Comment:
Shouldn't be included for the test cases below - since they're error tests we never get to the point where we check the value of the payload so it had no impact on the test, will delete.